install/kubernetes/helm/istio/charts/kiali/templates/service.yaml

```diff
@@ -3,15 +3,30 @@ kind: Service
 metadata:
   name: kiali
   namespace: {{ .Release.Namespace }}
+  annotations:
+    {{- range $key, $val := .Values.service.annotations }}
+    {{ $key }}: {{ $val | quote }}
+    {{- end }}
   labels:
     app: {{ template "kiali.name" . }}
     chart: {{ template "kiali.chart" . }}
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
 spec:
+  type: {{ .Values.service.type }}
   ports:
-  - name: http-kiali
+  - name: {{ .Values.service.name }}
+    targetPort: 20001
     protocol: TCP
-    port: 20001
+    port: {{ .Values.service.externalPort }}
   selector:
     app: kiali
+{{- if .Values.service.loadBalancerIP }}
+  loadBalancerIP: "{{ .Values.service.loadBalancerIP }}"
```

Comment:
As your reference, `type: LoadBalancer` should be added here in my test with metallb.